### PR TITLE
first redaction

### PR DESCRIPTION
**Pull request by stanmak1991**

### CODE REVIEW

File: src/com/company/ImmutablePerson.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
+package com.company;
+
+public class ImmutablePerson {
+    private String name;
+    private int age;
+    private boolean sex;
+
+    public ImmutablePerson(String name, int age, boolean sex) {
```
Comment (Siril-O) [inline src/com/company/ImmutablePerson.java]: Add ability to change one of the fields, so the class stays immutable

File: src/com/company/Main.java [hunk 2]
```diff
@@ -1,10 +1,28 @@
...
-	public static void main(String[] args) {
-		// TODO Auto-generated method stub
+import java.util.Arrays;
 
-	}
+public class Main {
 
-}
+    public static void main(String[] args) {
+        ImmutablePerson man = new ImmutablePerson("Alex", 22, true);
```
Comment (Siril-O) [inline src/com/company/Main.java]: create separate methods for each task testing, currently it is little  mess

File: src/com/company/Person.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+    public String getSurname() {
+        return surname;
+    }
+
+    public int getAge() {
+        return age;
+    }
+
+    public static Builder CreateBuilder() {
```
Comment (Siril-O) [inline src/com/company/Person.java]: lower case first letter

File: src/com/company/Person.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        return new Person().new Builder();
+    }
+
+    @Override
+    public String toString() {
+        return "BuilderPattern [id=" + id + ", name=" + name + ", surname=" + surname + ", age=" + age + "]";
```
Comment (Siril-O) [inline src/com/company/Person.java]: this is not BuilderPattern  it is Person

File: src/com/company/Person.java [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+    public String getSurname() {
+        return surname;
+    }
+
+    public int getAge() {
+        return age;
+    }
+
+    public static Builder CreateBuilder() {
+        return new Person().new Builder();
```
Comment (Siril-O) [inline src/com/company/Person.java]: Builder should create class at the end  to void not initialized Person

File: src/com/company/SortUtils.java [hunk 6]
```diff
@@ -0,0 +1,142 @@
+package com.company;
+
+public class SortUtils {
+    private static int Swap(int x, int y) {
```
Comment (Siril-O) [inline src/com/company/SortUtils.java]: why first letter upercase ?

File: src/com/company/SortUtils.java [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+        return x;
```
Comment (Siril-O) [inline src/com/company/SortUtils.java]: it does nothing

File: src/com/company/SortUtils.java [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+                ++i;
+                --y;
+            }
+        }
+        if (y > startIndex) quickSort(arr, startIndex, y);
+        if (i < lastIndex) quickSort(arr, i, lastIndex);
+        return 0;
+    }
+
+    public static int mergeSort(int[] arr) {
```
Comment (Siril-O) [inline src/com/company/SortUtils.java]: Split this giant method to smaller

File: src/com/company/SortUtils.java [hunk 9]
```diff
@@ -0,0 +1,142 @@
...
+            while (y > 0 && arr[y - 1] > sortItem) {
+                arr[y] = arr[y - 1];
+                y--;
+            }
+            arr[y] = sortItem;
+        }
+        return 0;
+    }
+
+    public static int SelectionSort(int[] arr) {
```
Comment (Siril-O) [inline src/com/company/SortUtils.java]: method names start with lower case

File: src/com/company/ImmutablePerson.java (hunk 1 above)
Comment (stanmak1991) [reply src/com/company/ImmutablePerson.java]: change

File: src/com/company/Person.java (hunk 3 above)
Comment (stanmak1991) [reply src/com/company/Person.java]: change

File: src/com/company/Person.java (hunk 4 above)
Comment (stanmak1991) [reply src/com/company/Person.java]: change

File: src/com/company/SortUtils.java (hunk 6 above)
Comment (stanmak1991) [reply src/com/company/SortUtils.java]: change

File: src/com/company/SortUtils.java (hunk 7 above)
Comment (stanmak1991) [reply src/com/company/SortUtils.java]: change

File: src/com/company/SortUtils.java (hunk 8 above)
Comment (stanmak1991) [reply src/com/company/SortUtils.java]: change

File: src/com/company/SortUtils.java (hunk 9 above)
Comment (stanmak1991) [reply src/com/company/SortUtils.java]: change

**Pull request closed without merging**